From: SeongJae Park <sj@kernel.org>
Date: Mon, 12 May 2025 10:26:23 -0700
Subject: [PATCH] mm/damon/stat: avoid <5 seconds interval stat updates

DAMON_STAT calcualtes and updates stat every aggregation interval.  The
auto-tuned interval allows it ranges from 100 milliseconds to 200
seconds.  100 milliseconds might be too small.  Make 5 seconds minimum
delay between stat updates.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/stat.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/mm/damon/stat.c b/mm/damon/stat.c
index 8351f6ac2ef9..75857dec03cb 100644
--- a/mm/damon/stat.c
+++ b/mm/damon/stat.c
@@ -111,6 +111,12 @@ static int damon_stat_after_aggregation(struct damon_ctx *c)
 	struct damon_target *t;
 	struct damon_region *r;
 	unsigned long access_bytes = 0;
+	static unsigned long last_refresh_jiffies = 0;
+
+	/* avoid unnecessarily frequent stat update */
+	if (time_before_eq(jiffies, last_refresh_jiffies +
+				msecs_to_jiffies(5 * MSEC_PER_SEC)))
+		return 0;
 
 	damon_for_each_target(t, c) {
 		damon_for_each_region(r, t)
-- 
2.39.5

